src/helpers/rgba.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import rgb from './rgb'
+
+type RgbaColor = {
+  red: number,
+  green: number,
+  blue: number,
+  alpha: number,
+}
+
+/**
+ * Returns a string value for the color. The returned result is the smallest possible rgba or hex notation.
+ *
+ * @example
+ * // Styles as object usage
+ * const styles = {
+ *   background: rgba(255, 205, 100, 180),
+ *   background: rgba({ red: 255, green: 205, blue: 100, alpha: 180 }),
+ *   background: rgba(255, 205, 100, 255),
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   background: ${rgba(255, 205, 100, 180)};
+ *   background: ${rgba({ red: 255, green: 205, blue: 100, alpha: 180 })};
+ *   background: ${rgba(255, 205, 100, 255)};
+ * `
+ *
+ * // CSS in JS Output
+ *
+ * element {
+ *   background: "rgba(255,205,100,180)";
+ *   background: "rgba(255,205,100,180)";
+ *   background: "#ffcd64";
+ * }
+ */
+function rgba(value: RgbaColor | number, green?: number, blue?: number, alpha?: number): string {
+  if (typeof value === 'number' &&
+      typeof green === 'number' &&
+      typeof blue === 'number' &&
+      typeof alpha === 'number') {
```

Comment:
From experience `null` or `undefined` often happen due a mistake. If we make alpha optional than the static analysis via Flowtype or TypeScript can't detect these errors and it would simply be opacity 1. Which also led me to another issue: what's the correct default? 1 or 0? It probably depends on the application.

On the other hand I'm a big fan of the robustness principle: "Be conservative in what you do, be liberal in what you accept from others" https://en.wikipedia.org/wiki/Robustness_principle

2 related questions:
1. If we set alpha to 1, should we also have default values for red, green, blue? Is it 0 or 255 or another value?
2. Should we check that red, green, blue are within the 0 - 255 boundary and alpha between 0 - 1?

Thoughts?